File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {

        List<Integer>[] bucket = new List[nums.length +1];
        HashMap<Integer, Integer> freq = new HashMap<>();
        for(int n: nums){
            freq.put(n,freq.getOrDefault(n,0) +1);
        }

        for(int key : freq.keySet()){
            int frequency = freq.get(key);
            if(bucket[frequency] == null){
                bucket[frequency] = new ArrayList<>();
            }   
            bucket[frequency].add(key);
        }

        int[] res = new int[k];

        int counter = 0;

        for(int i = bucket.length -1;i>=0  && counter < k;i--){
            if(bucket[i] != null){
                for(Integer integer : bucket[i]){
                    res[counter++] = integer;
                }
            }
        }

        return res;
        
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
var topKFrequent = function (nums, k) {


let map = new Map();

for(let num of nums){
    map.set(num, (map.get(num) || 0) + 1);
  }

  let freq =  Array.from(map.entries());

  freq.sort((a,b) => {
  return (b[1] - a[1]);
  });

  let res = [];

  for(let i=0;i<k;i++){
  res.push(freq[i][0]);
  }

  return res;





};
